src/render/markdown.rs

```diff
@@ -176,14 +179,14 @@ fn blend_fg_color(fg: SyntectColor, bg: SyntectColor) -> SyntectColor {
     if fg.a == 0xff {
         return fg;
     }
-    let ratio = fg.a as u32;
-    let r = (fg.r as u32 * ratio + bg.r as u32 * (255 - ratio)) / 255;
-    let g = (fg.g as u32 * ratio + bg.g as u32 * (255 - ratio)) / 255;
-    let b = (fg.b as u32 * ratio + bg.b as u32 * (255 - ratio)) / 255;
+    let ratio = u32::from(fg.a);
+    let r = (u32::from(fg.r) * ratio + u32::from(bg.r) * (255 - ratio)) / 255;
+    let g = (u32::from(fg.g) * ratio + u32::from(bg.g) * (255 - ratio)) / 255;
+    let b = (u32::from(fg.b) * ratio + u32::from(bg.b) * (255 - ratio)) / 255;
     SyntectColor {
-        r: r as u8,
-        g: g as u8,
-        b: b as u8,
+        r: u8::try_from(r).unwrap_or(u8::MAX),
+        g: u8::try_from(g).unwrap_or(u8::MAX),
+        b: u8::try_from(b).unwrap_or(u8::MAX),
```

Comment:
using as for conversion can be silently lossy, especially if you do further conversion later (which we do), but conversion functions such as u32::from will only perform lossless conversions.

https://rust-lang.github.io/rust-clippy/master/index.html#/cast_lossless

the u8::try_from... is to explicitly handle overflow (using as would just truncate it, but this makes it clear we took it into account)